core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -182,6 +201,21 @@ class CachedDataFrame(df: Dataset[Row], val queryString: String,
     }
   }
 
+  private[sql] def waitForLastShuffleCleanup(): Unit = {
```

Comment:
This was added to try waiting for cleanup of previous run of the cached plan so that it does not interferes with the current execution. The current execution should not wait for the cleanup synchronously.